presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1062,17 +1062,24 @@ public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTa
     {
         HiveTableLayoutHandle hiveLayoutHandle = checkType(layoutHandle, HiveTableLayoutHandle.class, "layoutHandle");
         List<ColumnHandle> partitionColumns = hiveLayoutHandle.getPartitionColumns();
-        List<TupleDomain<ColumnHandle>> partitionDomains = hiveLayoutHandle.getPartitions().get().stream()
-                .map(HivePartition::getTupleDomain)
-                .collect(toList());
+        List<HivePartition> partitions = hiveLayoutHandle.getPartitions().get();
 
-        TupleDomain<ColumnHandle> predicate = TupleDomain.none();
-        if (!partitionDomains.isEmpty()) {
-            predicate = TupleDomain.columnWiseUnion(partitionDomains);
```

Comment:
This is needed for predicate pushdown. Without it, the planner won't be able to derive tight predicates for the table scan.